File: CafeNet/Business Management/DTOs/CreatePaymentRequest.cs
using CafeNet.Data.Enums;

namespace CafeNet.Business_Management.DTOs;

public class CreatePaymentRequest
{
    public decimal TotalPrice { get; set; }
    public decimal UsedCredits { get; set; }
    public PaymentMethod Method { get; set; }
    public long UserId { get; set; }
    public long LocationId { get; set; }
    public long DiscountId { get; set; }

    public List<CreateOrderItemRequest> OrderItems { get; set; }
}

File: CafeNet.Front/src/types/dto/create-payment-request.ts
import { PaymentMethod } from "../enum/payment-method";
import { CreateOrderItemRequest } from "./create-order-request";

export interface CreatePaymentRequest {
  totalPrice: number;
  usedCredits: number;
  method: PaymentMethod;
  userId: number;
  locationId: number;
  discountId: number;
  orderItems: CreateOrderItemRequest[];
}
